asclepias_broker/harvester/metadata.py

```diff
@@ -110,25 +109,33 @@ class DOIMetadataHarvester(MetadataHarvester):
 
     DOI_ORG_AGENCY_API_URL = 'https://doi.org/doiRA'
 
-    def __init__(self, *, doi_api_url: str = None, resolvers: dict = None):
+    def __init__(self, *, doi_api_url: str = None, resolvers: dict = None,
+                 provider_name: str = None):
         """."""
         self.doi_api_url = doi_api_url or self.DOI_ORG_AGENCY_API_URL
         self.resolvers = resolvers or {
             'crossref': crossref_metadata,
             'datacite': datacite_metadata,
         }
+        self.provider_name = provider_name or "DOI metadata Harvester"
 
-    def can_harvest(self, identifier: str, scheme: str) -> bool:
+    def can_harvest(self, identifier: str, scheme: str,
+                    providers: List[str] = None) -> bool:
         """."""
-        return scheme.lower() == 'doi'
+        is_provider = False
+        if providers:
+            is_provider = self.provider_name in providers
+        return scheme.lower() == 'doi' and not is_provider
 
-    def harvest(self, identifier: str, scheme: str):
+    def harvest(self, identifier: str, scheme: str,
+                providers: List[str] = None):
         """."""
         data = self.get_metadata(identifier)
         if data:
+            providers.add(self.provider_name)
```

Comment:
Since providers is a list when passed initially to the function, you should cast to a `set`, add `self.provider_name`  and then turn back to a `list`.